azurelinuxagent/common/event.py

```diff
@@ -640,6 +640,34 @@ def report_periodic(delta, op, is_success=True, message=''):
                  op=op)
 
 
+def report_periodic_metric(delta, category, counter, instance, value, log_event=False, reporter=__event_logger__):
```

Comment:
reporting for periodic metrics.

Reply:
all metrics are reported periodically, it's just that the period is different. i think we need a different implementation than checking for report_period to be None, which is confusing

Reply:
I introduced new flag for checking, hopefully that avoids confusion.